kernel/src/engine/arrow_expression/mod.rs

```diff
@@ -213,6 +213,24 @@ impl Scalar {
     }
 }
 
+impl ArrayData {
+    /// Convert kernel [`ArrayData`] to Arrow [`ArrayRef`].
+    pub fn to_arrow(&self) -> DeltaResult<ArrayRef> {
```

Comment:
The existing `Scalar.to_array()` function produces a `ListArray` with _n_ `PrimitiveArrays` of size _n_ for an _n_-size `ArrayData`. Therefore, I added this dedicated function to convert `ArrayData` into a single `PrimitiveArray` of size _n_.